### PR TITLE
Master build

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Builds
Builds MSIs
Builds bundle
Builds swix

.NET Core is not signed. Will wait until @int19h merges msys

Note that there is a workaround for VSIX build issue. VSIX does not like multi-targeted .NET Core projects as references since they don't have certain targets. Thus we have to add empty targets to enable VSIX build and F5 scenario in debug. However, said empoty targets prevent RTVS package files from being copied over - so there is task that does it manually in Debug.

In Release we do not need VSIX since we build swix and thus we don't build it and copying works normally. F5 would be broken though.

### CODE REVIEW

File: nuget.config [hunk 1]
```diff
@@ -3,4 +3,15 @@
   <config>
     <add key="repositoryPath" value="NugetPackages" />
   </config>
+  <activePackageSource>
+    <add key="All" value="(Aggregate source)" />
+  </activePackageSource>
+  <packageSources>
+    <add key="NuGet V3" value="https://api.nuget.org/v3/index.json" />
+    <add key="VS Internal" value="https://mseng.pkgs.visualstudio.com/DefaultCollection/_packaging/VS/nuget/v3/index.json" />
```
Comment (AlexanderSher) [inline nuget.config]: I think we can add internals to `RTVS-pr` nuget config

File: src/Common/Core/Impl/Microsoft.Common.Core.csproj [hunk 2]
```diff
@@ -28,11 +38,13 @@
...
   </ItemGroup>
+  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
   <Import Project="$(SourceDirectory)R.Build.Version.targets" />
-  <Import Project="$(SourceDirectory)R.Build.Loc.targets" />
-  <Target Name="BuiltProjectOutputGroup" Condition=" '$(TargetFramework)' == 'net46' " />
-  <Target Name="BuiltProjectOutputGroupDependencies" Condition=" '$(TargetFramework)' == 'net46' " />
-  <Target Name="GetCopyToOutputDirectoryItems" Condition=" '$(TargetFramework)' == 'net46' " />
-  <Target Name="SatelliteDllsProjectOutputGroup" Condition=" '$(TargetFramework)' == 'net46' " />
-  <Target Name="DebugSymbolsProjectOutputGroup" Condition=" '$(TargetFramework)' == 'net46' " />
+  <Import Project="$(SourceDirectory)R.Workarounds.targets" Condition="'$(Configuration)'=='Debug'"/>
```
Comment (AlexanderSher) [inline src/Common/Core/Impl/Microsoft.Common.Core.csproj]: Can we move `Condition="'$(Configuration)'=='Debug'"` to targets file itself?

File: src/R.Build.Version.targets [hunk 3]
```diff
@@ -1,11 +1,15 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup >
     <AssemblyVersionInfoPath>$(MSBuildProjectDirectory)\AssemblyVersionInfo.cs</AssemblyVersionInfoPath>
-    <GenerateAssemblyVersionInfo Condition="'$(GenerateAssemblyVersionInfo)' == '' AND Exists('$(AssemblyVersionInfoPath)')">false</GenerateAssemblyVersionInfo>
-    <GenerateAssemblyVersionInfo Condition="'$(GenerateAssemblyVersionInfo)' == '' AND !Exists('$(AssemblyVersionInfoPath)')">true</GenerateAssemblyVersionInfo>
+    <!-- Suppress automatic generation of version information. We generate it ourselves
+      since we need it for both .NET Core and .NET Desktop -->
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <GenerateAssemblyVersionInfo>true</GenerateAssemblyVersionInfo>
+    <!--<GenerateAssemblyVersionInfo Condition="'$(GenerateAssemblyVersionInfo)' == '' AND Exists('$(AssemblyVersionInfoPath)')">false</GenerateAssemblyVersionInfo>
```
Comment (AlexanderSher) [inline src/R.Build.Version.targets]: Do you remember what was the purpose of these conditional values?

File: src/R/Components/Impl/Microsoft.R.Components.csproj [hunk 4]
```diff
@@ -1,9 +1,16 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project>
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
```
Comment (AlexanderSher) [inline src/R/Components/Impl/Microsoft.R.Components.csproj]: Are you sure we need this?

File: src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs [hunk 5]
```diff
@@ -1,19 +0,0 @@
...
-using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
-using System.Linq;
-using Microsoft.Markdown.Editor.Test;
-using Microsoft.R.Editor.Test.Fixtures;
-
-namespace Microsoft.R.Editor.Application.Test.Fixtures {
-    // Fixture doesn't import itself. Use AssemblyFixtureImportAttribute
-    [ExcludeFromCodeCoverage]
-    public class REditorApplicationServicesFixture : MarkdownEditorServicesFixture {
```
Comment (AlexanderSher) [inline src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs]: I thought we have exports in these assemblies

File: src/Windows/Setup/WixInstall/WixInstall.csproj [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (AlexanderSher) [inline src/Windows/Setup/WixInstall/WixInstall.csproj]: What is this project for?

File: src/R/Support/Impl/Microsoft.R.Support.csproj [hunk 7]
```diff
@@ -1,203 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (AlexanderSher) [inline src/R/Support/Impl/Microsoft.R.Support.csproj]: What has happened to the content of this project?

File: src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs (hunk 5 above)
Comment (MikhailArkhipov) [inline src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs]: Guess may need to merge master

File: src/R.Build.Version.targets (hunk 3 above)
Comment (MikhailArkhipov) [inline src/R.Build.Version.targets]: Not to generate version info in debug builds. I'll restore this.

File: src/Windows/Setup/WixInstall/WixInstall.csproj (hunk 6 above)
Comment (MikhailArkhipov) [inline src/Windows/Setup/WixInstall/WixInstall.csproj]: To install Wix tools from nuget

File: src/R/Support/Impl/Microsoft.R.Support.csproj (hunk 7 above)
Comment (MikhailArkhipov) [inline src/R/Support/Impl/Microsoft.R.Support.csproj]: Gone to R.Editor. HTML part was not needed and RData went to the editor

File: src/Common/Core/Impl/Microsoft.Common.Core.csproj (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Microsoft.Common.Core.csproj]: Sure

File: nuget.config (hunk 1 above)
Comment (MikhailArkhipov) [inline nuget.config]: I think this is for Microbuild stuff. mseng == MB.

File: src/Windows/Setup/WixInstall/WixInstall.csproj (hunk 6 above)
Comment (MikhailArkhipov) [inline src/Windows/Setup/WixInstall/WixInstall.csproj]: There is no 'restore' target in wixproj

File: src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs (hunk 5 above)
Comment (MikhailArkhipov) [inline src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs]: Not sure why this is different, should be same as master... R.Support was removed back in my previous merge (large editor split PR)

File: src/R/Support/Impl/Microsoft.R.Support.csproj (hunk 7 above)
Comment (MikhailArkhipov) [inline src/R/Support/Impl/Microsoft.R.Support.csproj]: I think I know why it shows. in previous PR project was removed from solution but not actually deleted. https://github.com/Microsoft/RTVS/tree/master/src/R/Support still there in master

File: src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs (hunk 5 above)
Comment (MikhailArkhipov) [inline src/R/Editor/Application.Test/Fixtures/REditorApplicationServicesFixture.cs]: Looks like file was removed from the project but not deleted

**Comment by int19h** [comment]
Can you try building with the changes that I have just merged? It builds VSIX fine for me from the merged branch, so whatever workarounds were necessary should already be in there now.

**Comment by int19h** [comment]
Also take a look at this.
https://github.com/Microsoft/RTVS-pr/compare/msys?expand=1

**Comment by MikhailArkhipov** [comment]
Doing it as we speak

**Comment by MikhailArkhipov** [comment]
host.exe appears in C:\RTVS-Main\src\Host\Process\bin\Release -> rather than in C:\RTVS-Main\bin\Release

**Comment by int19h** [comment]
Are you building the host via MSBuild? CMake will output to project directory by default, but .vcxproj should invoke the build script with appropriate arguments to put it in solution output dir.

**Comment by MikhailArkhipov** [comment]
That was in the lab. I'll try rebuilding

**Comment by MikhailArkhipov** [comment]
Built successfully, including setup and swix

**Comment by MikhailArkhipov** [comment]
If no objections, will merge so we can kick nightly builds. Will sort out signing then.

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**